internal/config/config.go

```diff
@@ -772,36 +772,41 @@ func (c Config) GetKVS(s string, defaultKVS map[string]KVS) (Targets, error) {
 }
 
 // DelKVS - delete a specific key.
-func (c Config) DelKVS(s string) error {
-	if len(s) == 0 {
-		return Errorf("input arguments cannot be empty")
-	}
-	inputs := strings.Fields(s)
-	if len(inputs) > 1 {
-		return Errorf("invalid number of arguments %s", s)
-	}
-	subSystemValue := strings.SplitN(inputs[0], SubSystemSeparator, 2)
-	if len(subSystemValue) == 0 {
-		return Errorf("invalid number of arguments %s", s)
-	}
-	if !SubSystems.Contains(subSystemValue[0]) {
+func (c Config) DelKVS(s string, defaultKVS map[string]KVS) error {
+	subSys, inputs, tgt, err := GetSubSys(s)
+	if !SubSystems.Contains(subSys) {
 		// Unknown sub-system found try to remove it anyways.
-		delete(c, subSystemValue[0])
+		delete(c, subSys)
 		return nil
 	}
-	tgt := Default
-	subSys := subSystemValue[0]
-	if len(subSystemValue) == 2 {
-		if len(subSystemValue[1]) == 0 {
-			return Errorf("sub-system target '%s' cannot be empty", s)
-		}
-		tgt = subSystemValue[1]
+	if err != nil {
+		return err
 	}
-	_, ok := c[subSys][tgt]
+
+	ck, ok := c[subSys][tgt]
 	if !ok {
-		return Errorf("sub-system %s already deleted", s)
+		return Errorf("sub-system %s already deleted", inputs[0])
```

Comment:
This needs to be improved- maybe:
```suggestion
		return Errorf("sub-system %s:%s does not exist", subSys, tgt)
```

Reply:
I believe this is supposed to indicate that the subsystem has been reset but the server has yet to be restarted, like if you were to call `mc admin config reset site` it would delete it, but if you were to try that same command again before restarting the server it would return that error. Is "does not exist" accurate for that?

Reply:
Changed it to `return Errorf("sub-system %s:%s already deleted or does not exist", subSys, tgt)`, does this work better?